### PR TITLE
Added list of top reviewers

### LINKED ISSUE #11: Add top reviewers
Add a new metric which shows who are the top reviewers in a PR.

This could "gamify" the review experience.

### PR DESCRIPTION
**Pull request by Bullrich**
Added list of top reviewers per month.

Added three new reports:
- a list showing who reviewed the most PRs each month.
- Who is the current "reviewer of the month" (who made the biggest amount of reviews in the last month).
- Who is the **top** reviewer of the repo (who made the biggest amount of reviews)

Resolves paritytech/metrics#11

# Summary

- **New Features**
	- Support for obtaining metrics from Issues in addition to Pull Requests.
	- Display of author avatars in Pull Requests.
	- Inclusion of new metrics in reports: Reviewers' statistics, Top reviewer, and Reviewer of the month.

- **Enhancements**
	- Improved visibility of author information in Pull Requests.
	- Enhanced reporting capabilities with additional reviewer metrics and visual charts.

- **Documentation**
	- Updated README to reflect the new metrics support.

### CODE REVIEW

**Comment by Bullrich** [comment]
Find walkthrough of the code here:
- https://github.com/paritytech-stg/metrics/pull/2#issuecomment-1961696609

File: src/report/pullRequests.ts [hunk 1]
```diff
@@ -148,4 +159,113 @@ export class PullRequestAnalytics {
 
     return averages;
   }
+
+  /** Returns the reviewer who gave the biggest amount of reviews per month */
+  getMonthlyReviewers(
```
Comment (rzadp) [inline src/report/pullRequests.ts]: Would a better name be `getTopMonthlyReviewers`?

**Review by rzadp — APPROVED** [review]

File: src/report/pullRequests.ts [hunk 2]
```diff
@@ -148,4 +159,113 @@ export class PullRequestAnalytics {
...
+          avatar: review.author.avatarUrl,
+        });
+      }
+    }
+
+    return monthsWithMatches;
+  }
+
+  /** Returns the reviewer who gave the biggest amount of reviews */
+  getTopReviewer(
```
Comment (mordamax) [inline src/report/pullRequests.ts:243]: topReviewer within which timeframe?

Comment (Bullrich) [reply src/report/pullRequests.ts:243]: In total.

We have two reviewers, the top of the month and the top in the repository's history.

Here is an example, where I'm the month's top reviewer, but you @mordamax are the top reviewer in the repo's history:
<details>
<summary>Opstooling metrics</summary>

<h3>Top reviewers</h3>

```mermaid
  gantt
    title Top reviewer per month
    dateFormat X
    axisFormat %s
    section Mar 2022
    mordamax : 0, 6
    section Apr 2022
    mordamax : 0, 1
    section May 2022
    mordamax : 0, 2
    section Jul 2022
    cmichi : 0, 2
    section Aug 2022
    cmichi : 0, 3
    section Sep 2022
    mordamax : 0, 1
    section Oct 2022
    BulatSaif : 0, 1
    section Jan 2023
    rzadp : 0, 17
    section Feb 2023
    mordamax : 0, 8
    section Mar 2023
    mordamax : 0, 2
    section Apr 2023
    mordamax : 0, 21
    section May 2023
    Bullrich : 0, 8
    section Jun 2023
    Bullrich : 0, 9
    section Jul 2023
    mutantcornholio : 0, 15
    section Aug 2023
    mordamax : 0, 8
    section Sep 2023
    mordamax : 0, 2
    section Oct 2023
    Bullrich : 0, 2
    section Nov 2023
    rzadp : 0, 1
    section Dec 2023
    Bullrich : 0, 2
    section Jan 2024
    Bullrich : 0, 2
  ```
<h4>Reviewer of the month</h4>
Bullrich's avatar
@Bullrich with 2 reviews!
<h3>Top reviewer</h3>

mordamax's avatar

@mordamax with a **total of 84 reviews**!<h1>Issues</h1>
</details>

Comment (mordamax) [reply src/report/pullRequests.ts:243]: @Bullrich does it count the approvals or comments as well?
 I guess if it includes comments - then im testing a lot the command bot, which might have impacted that, but I'm not reviewing that much i think...

Comment (Bullrich) [reply src/report/pullRequests.ts:243]: It counts the review action, but not the comments in the review:
image

**Pull request merged**